src/PSRule.Rules.Azure/rules/Azure.MySQL.Rule.ps1

```diff
@@ -87,6 +87,23 @@ Rule 'Azure.MySQL.AAD' -Ref 'AZR-000392' -Type 'Microsoft.DBforMySQL/flexibleSer
     }
 }
 
+# Synopsis: Ensure Azure AD-only authentication is enabled with Azure Database for MySQL databases.    
+Rule 'Azure.MySQL.AADOnly' -Ref 'AZR-000394' -Type 'Microsoft.DBforMySQL/flexibleServers', 'Microsoft.DBforMySQL/flexibleServers/configurations' -Tag @{ release = 'GA'; ruleSet = '2023_09'; 'Azure.WAF/pillar' = 'Security'; } -Labels @{ 'Azure.MCSB.v1/control' = 'IM-1' } {
+    if ($PSRule.TargetType -eq 'Microsoft.DBforMySQL/flexibleServers') {
+        $configurations = @(GetSubResources -ResourceType 'Microsoft.DBforMySQL/flexibleServers/configurations' -Name "aad_auth_only")
+        if ($configurations.Count -eq 0) {
+            return $Assert.Fail().Reason($LocalizedData.SubResourceNotFound, 'Microsoft.DBforMySQL/flexibleServers/configurations')
+        }
+
+        foreach ($config in $configurations) {
+            $Assert.HasFieldValue($config, 'properties.currentValue', 'ON')
+        }
+    }
+    elseif ($PSRule.TargetName -cmatch 'aad_auth_only') {
+        $Assert.HasFieldValue($TargetObject, 'properties.currentValue', 'ON')
+    }
```

Comment:
We need to handle the case where the configuration value is not called `aad_auth_only`.
```suggestion
    }
    else {
      $Assert.Pass()
    }
```